### PR TITLE
Update ModelCheckpoint callback in callbacks.py

### PR DESCRIPTION
**Pull request by 8bitmp3**
See #29958 : Suggested improvements:

* _Clarified and expanded the ModelCheckpoint callback description_:
`ModelCheckpoint` does not just "save after each epoch" - it can be customized.

* _Add a ModelCheckpoint example_:
An example showing how to save checkpoints with weights only after 1 epoch and load weights for a new model with `tf.train.latest_checkpoint`. Used the following tutorial as a reference: _Save and Restore Models_ [r1.14](https://www.tensorflow.org/tutorials/keras/save_and_restore_models) and [r2.0](https://www.tensorflow.org/beta/tutorials/keras/save_and_restore_models)

Any questions, suggestions, fixes, clarifications welcome!

### CODE REVIEW

**Comment by 8bitmp3** [comment]
@rthadur Hi, just checking if this is still alright. Thanks a million

File: tensorflow/python/keras/callbacks.py [hunk 1]
```diff
@@ -788,15 +788,54 @@ def on_epoch_end(self, epoch, logs=None):
...
+  
+  This callback allows to use the trained model without having to retrain it or 
+  pick up training where you left off in the event of, for example, an interrupted 
+  training process.
 
-  `filepath` can contain named formatting options,
-  which will be filled the value of `epoch` and
-  keys in `logs` (passed in `on_epoch_end`).
+  A path to save the model path - `filepath` - can contain named formatting options, 
+  which will be filled the value of `epoch` andkeys in `logs` (passed in `on_epoch_end`).
```
Comment (rchao) [inline tensorflow/python/keras/callbacks.py]: space between "and keys"

File: tensorflow/python/keras/callbacks.py [hunk 2]
```diff
@@ -788,15 +788,54 @@ def on_epoch_end(self, epoch, logs=None):
...
+  checkpoint_dir = os.path.dirname(checkpoint_path)
+
+  # Create a checkpoint callback, save weights only after each epoch with `save_freq='epoch'
+  # Alternatively, instead of `save_freq` you can use `period={N_integer} to save after every N epoch)
+
+  cp_callback = tf.keras.callbacks.ModelCheckpoint(filepath=checkpoint_path, 
+                                                   save_weights_only=True, 
+                                                   verbose=1,
+                                                   save_freq='epoch'
+                                                   )
```
Comment (rchao) [inline tensorflow/python/keras/callbacks.py]: looks to me this can be with the previous line

**Review by rchao — COMMENTED** [review]
Thanks for the request!

File: tensorflow/python/keras/callbacks.py [hunk 3]
```diff
@@ -788,15 +788,54 @@ def on_epoch_end(self, epoch, logs=None):
 
 @keras_export('keras.callbacks.ModelCheckpoint')
 class ModelCheckpoint(Callback):
-  """Save the model after every epoch.
+  """Save the model checkpoints during and at the end of training.
+  
+  This callback allows to use the trained model without having to retrain it or
```
[the remark this answers is not in the document]
Comment (8bitmp3) [reply tensorflow/python/keras/callbacks.py]: Cool. I simplified it to be more human friendly:
```
The current weights of the model are saved continuously during training. 
These checkpoint(s) can be loaded later into a new model, for example.
```

File: tensorflow/python/keras/callbacks.py (hunk 2 above)
Comment (8bitmp3) [reply tensorflow/python/keras/callbacks.py]: Ok. A few less lines now.

File: tensorflow/python/keras/callbacks.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (8bitmp3) [reply tensorflow/python/keras/callbacks.py]: Also added:
```
Optionally, use the `save_best_only=True` argument to save the model with the best performance.
```

**Comment by 8bitmp3** [comment]
> Thanks for the request!

Revised, thank you @rchao 👍

**Comment by mellanox-github** [comment]
Can one of the admins verify this patch?

**Review by cpoptic — APPROVED** [review]

**Comment by gbaned** [comment]
@8bitmp3 Could you please check build failures and resolve the conflicts? Thanks!

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this PR still valid? Assigning the `stalled` label. Please comment to reassure me that this is still being worked on.
